matminer/utils/data.py

```diff
@@ -155,6 +188,54 @@ def __init__(self):
             "sat_magn",
         ]
 
+        self.impute_nan = impute_nan
```

Comment:
Just a general comment on `impute_nan`. Having it as a bool arg is somewhat restrictive if in future a new imputation mechanism was added, e.g., geometric mean instead of arithmetic, logarithmic or whatever. Do you think it is worth converting the arg to an optional string with possible values `[None, arithmetic_mean]` at this stage? Maybe this is an overcomplication...

Reply:
I agree that it would be best to convert the arg to a string, however, this imputation will lead to different feature values for a few data points only (hopefully) and I expect the way the imputation is done will not drastically change a model performances, as long as it's done not-too-stupidly. I'm not even 100% sure all this impute_nan stuff is not already overthinking things...

Reply:
I guess I don't have any problem with adding it as a bool for now, it could still be extended such that `True` means "`mean`" and and any other strings in the future have their own thing. Let's leave it as-is.